core/chapters/c12_dictionaries.py

```diff
@@ -0,0 +1,510 @@
+
+    __copyable__
+    __no_auto_translate__
+
+    quantities = {'apple': 1, 'cat': 10}
+    print(quantities)
+
+
+        """
+
+        program = """
+        quantities = {'apple': 1, 'cat': 10}
+        print(quantities.keys())
+        """
+
+        def check(self):
+            return "dict_keys(['apple', 'cat'])" in self.result
+
+
+    class keys_are_iterable(VerbatimStep):
+        """
+The `.keys()` method of `dict` does basically what you'd expect. You can iterate over the value it returns
+just like you'd iterate over a list:
+
+    __copyable__
+    __program_indented__
+        """
+
+        def program(self):
+            quantities = {'apple': 1, 'cat': 10}
+            for key in quantities.keys():
+                print(key)
+
+    class keys_are_iterable2(Step):
+        """
+Actually, you don't even need `.keys()`. Iterating directly over a dictionary automatically iterates over its keys.
+Sometimes it's nice to write `.keys()` to make your code more readable, but you don't have to.
+Remove the `.keys()` and run the code again.
+        """
+
+        program = """
+        quantities = {'apple': 1, 'cat': 10}
+        for key in quantities:
+            print(key)
+                """
+
+        def check(self):
+            return "apple\ncat" in self.result
+
+    class cleanup_shopping_cart(ExerciseStep):
+        """
+Now you can use this to modify our function on the previous page to remove the `cart` argument:
+
+    __copyable__
+    __no_auto_translate__
+
+    def total_cost(quantities, prices):
+        result = 0
+        for item in ...:
+            price = prices[item]
+            quantity = quantities[item]
+            result += price * quantity
+        return result
+
+    assert_equal(
+        total_cost(
+            {'dog': 5000000, 'box': 2},
+            {'apple': 2, 'box': 5, 'cat': 100, 'dog': 100},
+        ),
+        500000010,
+    )
+        """
+
+        def solution(self):
+            def total_cost(quantities: Dict[str, str], prices: Dict[str, str]):
+                result = 0
+                for item in quantities:
+                    price = prices[item]
+                    quantity = quantities[item]
+                    result += price * quantity
+                return result
+
+            return total_cost
+
+        hints = "TODO PPPP"
+
+        tests = [
+          (
+                ({'apple' : 3, 'carrot' : 4}, {'apple' : 10, 'carrot' : 20}),
+                110
+           )
+        ]
+
+    class english_to_french(ExerciseStep):
+        """
+That looks nice! We've fully solved the problem of adding up the total cost.
+
+Coming back to our first example: write a function
+which prints out each word in an English-to-French dictionary and its translation, labeling them with their languages.
+Here's your starting code:
+
+    __copyable__
+    __no_auto_translate__
```

Comment:
Even here, I think automatic translation will be fine. I expect the French translations for each word to be:

```
french -> anglais
apple -> pomme
pomme -> apple
box -> boite
boite -> box
```